glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -152,7 +157,26 @@ open class GleanInternalAPI internal constructor () {
      */
     fun setUploadEnabled(enabled: Boolean) {
         if (isInitialized()) {
-            LibGleanFFI.INSTANCE.glean_set_upload_enabled(handle, enabled.toByte())
+            val originalEnabled = getUploadEnabled()
+
+            Dispatchers.API.launch {
+                // glean_set_upload_enabled might delete all of the queued pings. We
+                // therefore need to obtain the lock from the PingUploader so that
+                // iterating over and deleting the pings doesn't happen at the same time.
+                synchronized(PingUploader.pingQueueLock) {
+                    LibGleanFFI.INSTANCE.glean_set_upload_enabled(handle, enabled.toByte())
+                }
+            }
+
+            Dispatchers.API.launch {
```

Comment:
Couldn't we merge the two dispatcher blocks into one?

Reply:
You're totally right.